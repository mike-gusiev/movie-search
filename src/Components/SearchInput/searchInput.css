.search {
    display: flex;
}

.search__input {
    padding: 1rem;
    font-size: 2rem;
    width: 100%;
    border: 5px solid black;
    border-radius: 20px;
    outline: none;
}