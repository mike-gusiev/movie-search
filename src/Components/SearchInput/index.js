import React from 'react';
import PropTypes from 'prop-types';

import './searchInput.css'

const SearchInput = ({value, onChange}) => (
    <div className="search">
        <input
            placeholder='Enter movie name'
            className="search__input"
            value={value}
            onChange={(e) => onChange(e)}/>
    </div>
);

export default SearchInput;

SearchInput.propTypes = {
    value: PropTypes.string,
    onChange: PropTypes.func
};
