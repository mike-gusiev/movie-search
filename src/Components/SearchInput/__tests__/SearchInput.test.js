import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import SearchInput from '../index';

const defaultProps = {
    value : 'search',
    onChange : jest.fn()
};

const setup = (props = defaultProps) => {
    const wrapper = shallow(<SearchInput {...props} />);
    return {
        wrapper,
        props
    };
};

describe('TEST Search input component ', () => {

    test('TEST Should match the snapshot.', () => {
        const { wrapper } = setup();
        expect(toJson(wrapper)).toMatchSnapshot();
    });

});
