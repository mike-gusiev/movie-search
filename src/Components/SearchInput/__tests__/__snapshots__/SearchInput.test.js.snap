// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST Search input component  TEST Should match the snapshot. 1`] = `
<div
  className="search"
>
  <input
    className="search__input"
    onChange={[Function]}
    placeholder="Enter movie name"
    value="search"
  />
</div>
`;
